lang/cs/Org.Apache.REEF.Client/Common/JobSubmissionResultImpl.cs

```diff
@@ -43,20 +45,57 @@ internal class HttpClientHelper : IDriverHttpEndpoint
         private const string AppJson = "application/json";
 
         private string _driverUrl;
+        private string _appId;
 
         private readonly HttpClient _client;
+        private readonly IREEFClient _reefClient;
 
-        internal HttpClientHelper()
+        internal JobSubmissionResultImpl(IREEFClient reefClient, string filePath, bool yarn)
         {
+            _reefClient = reefClient;
             _client = new HttpClient
             {
                 Timeout = TimeSpan.FromSeconds(SecondsForHttpClientTimeout),
             };
             _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(AppJson));
+
+            if (yarn)
```

Comment:
Totally agree. I was trying not to make too dramatic changes on the original code as it will change. For now, I introduced an abstract method GetDriverUri() at base level and allow each runtime to implement its own.